src/webhook/mutation/pod_mutator/oneagent_mutation/mutator.go

```diff
@@ -0,0 +1,95 @@
+package oneagent_mutation
+
+import (
+	dynatracev1beta1 "github.com/Dynatrace/dynatrace-operator/src/api/v1beta1"
+	"github.com/Dynatrace/dynatrace-operator/src/initgeneration"
+	"github.com/Dynatrace/dynatrace-operator/src/kubeobjects"
+	dtwebhook "github.com/Dynatrace/dynatrace-operator/src/webhook"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type OneAgentPodMutator struct {
+	image            string
+	clusterID        string
+	webhookNamespace string
+	client           client.Client
+	apiReader        client.Reader
+}
+
+func NewOneAgentPodMutator(image, clusterID, webhookNamespace string, client client.Client, apiReader client.Reader) *OneAgentPodMutator {
+	return &OneAgentPodMutator{
+		image:            image,
+		clusterID:        clusterID,
+		webhookNamespace: webhookNamespace,
+		client:           client,
+		apiReader:        apiReader,
+	}
+}
+
+func (mutator *OneAgentPodMutator) Enabled(request *dtwebhook.BaseRequest) bool {
+	return kubeobjects.GetFieldBool(request.Pod.Annotations, dtwebhook.AnnotationOneAgentInject, true)
+}
+
+func (mutator *OneAgentPodMutator) Injected(request *dtwebhook.BaseRequest) bool {
+	return kubeobjects.GetFieldBool(request.Pod.Annotations, dtwebhook.AnnotationOneAgentInjected, false)
+}
+
+func (mutator *OneAgentPodMutator) Mutate(request *dtwebhook.MutationRequest) error {
+	log.Info("injecting OneAgent into pod", "pod", request.Pod.GenerateName)
+	if err := mutator.ensureInitSecret(request); err != nil {
+		return errors.WithStack(err)
+	}
+
+	installerInfo := getInstallerInfo(request.Pod)
+	mutator.addVolumes(request.Pod, request.DynaKube)
+	mutator.configureInitContainer(request, installerInfo)
+	mutator.mutateUserContainers(request)
+	setInjectedAnnotation(request.Pod)
+	return nil
+}
+
+func (mutator *OneAgentPodMutator) Reinvoke(request *dtwebhook.ReinvocationRequest) bool {
+	if !mutator.Injected(request.BaseRequest) {
+		return false
+	}
+	log.Info("reinvoking", "pod", request.Pod.GenerateName)
```

Comment:
this logs the pod name, but is done per mutator.
could be either moved up where the different mutators are called, or log what type of mutator is reinvoking.

Reply:
its done per mutator so in the log line you can see where it came from (different package ==> different logger)